routes/article.js

```diff
@@ -65,34 +65,31 @@ function dropArticles() {
 function updateNews() {
     // hourly
     const url = `${process.env.NEWS_URI}`
-    request(url, (err, res, body) => {
-        if (!err && res.statusCode == 200) {
-            const headlines = JSON.parse(body).articles
+    request(url)
+        .then((res) => {
+            const headlines = JSON.parse(res).articles
             let articles = headlines.map((old) => {
                 // save art
-                Article.find({url: old.url}).then((found) => {
-                    if (!found.length) {
-                        let art = new Article()
-                        Object.keys(old).forEach((key) => {
-                            art[key] = old[key]
-                        })
-                        art['keywords'] = getPhrases(art)
-                        art['votes'] = 0
+                Article.findOne({url: old.url})
+                    .then((article) => {
+                        // already in database, do nothing
+                        if (!article) throw new Error('Article not found')
+                    })
+                    .catch((err) => {
+                        // not in db
+                        let artObj = Object.assign({
```

Comment:
Why create a new object? What about added properties to the old one and then `new Article(old)`?